app/src/main/java/com/example/foodmanchu/RecipesFragment.kt

```diff
@@ -10,6 +10,10 @@ class RecipesFragment: Fragment(R.layout.recipes_fragment) {
         super.onCreate(savedInstanceState)
         val binding = RecipesFragmentBinding.bind(view)
 
+        binding.addRecipeFab.setOnClickListener {
+            AddRecipeDialog().show(parentFragmentManager,"Open Add Recipe")
```

Comment:
this sounds strange... no comment on it other than that.

Reply:
yeah this one is wierd , but it works, its a multi selection alert dialog, that's where ingredients are added to a recipe, it shows them in a textview then I build a string based on the textview.text and set the object parameter to that. as well as other stuff in the other dialog lol